build-logic/gradle/src/main/kotlin/com/avito/android/GenerateCommonProperties.kt

```diff
@@ -0,0 +1,55 @@
+package com.avito.android
+
+import org.apache.commons.configuration.PropertiesConfiguration
+import org.gradle.api.DefaultTask
+import org.gradle.api.file.Directory
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.provider.SetProperty
+import org.gradle.api.tasks.InputFile
+import org.gradle.api.tasks.OutputDirectories
+import org.gradle.api.tasks.TaskAction
+import java.io.File
+
+abstract class GenerateCommonProperties : DefaultTask() {
+
+    @get:InputFile
+    abstract val commonPropertiesFile: RegularFileProperty
+
+    @get:OutputDirectories
+    abstract val projectDirs: SetProperty<Directory>
+
+    @TaskAction
+    fun doWork() {
+        val commonProperties = loadProperties(commonPropertiesFile.get().asFile)
+
+        projectDirs.get().forEach { projectDir ->
+
+            val gradlePropertiesFile = File(projectDir.asFile, "gradle.properties")
+
+            val properties = if (gradlePropertiesFile.exists()) {
+                loadProperties(gradlePropertiesFile)
+            } else {
+                PropertiesConfiguration(gradlePropertiesFile)
+            }
+
+            commonProperties.keys.forEach { key ->
+                properties.setProperty(key, commonProperties.getProperty(key))
```

Comment:
Proposal

- [ ] Add the comment `#overriden` on each overridden property
- [ ] Add the comment `#from common file` on each property from common file
This will help to understand faster where I should add or update the property

Reply:
Good idea! Will do

Reply:
`#overriden` it's when for example you have property `org.gradle.xxx`
and the common file has a `org.gradle.xxx=1`
and some project `build.properties` has the same `org.gradle.xxx` prop
As I understand, after generation project's `build.properties` will have  `org.gradle.xxx` with value from common file

Reply:
yep, but it will be with `#from common file` comment now, why `#overriden`?

Reply:
![image](https://user-images.githubusercontent.com/1105133/129551542-9751d8ab-4ee1-4a2b-b745-a2cd70c51442.png)

looks like that. ok?